### ISSUE TITLE
Port twisted.internet._sslverify to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6141]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6141|
|Type|enhancement|
|Created|2012-10-16 19:21:07Z|
|Branch|https://github.com/twisted/twisted/tree/sslverify-py3-6141-2|

`twisted.internet._sslverify` should run on Python 3.

(Possibly unnecessary for initial milestone.)

<details><summary>Searchable metadata</summary>

```
trac-id__6141 6141
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_sslverify_py3_6141_2 branches/sslverify-py3-6141-2
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1350415267000000 1350415267000000
changetime__1351616820000000 1351616820000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [36245]) Branching to 'sslverify-py3-6141'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst commented|
|-|-|

(In [36252]) Branching to 'sslverify-py3-6141-2'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst set owner to @exarkun|
|-|-|

OK, did enough of a port that I think we can use this as a dependency.

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/sslverify-py3-6141-2 is running.

[#6156](https://github.com/twisted/twisted/issues/6156) opened for missing tests.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @itamarst|
|-|-|

Thanks.

  1. `twisted/internet/_sslverify.py`
    1. Can you nuke the duplicate TML copyright header?
    1. `DistinguishedName` looks like it needs some better documentation about its initializer arguments and attributes (particularly the types of their values).
  1. `twisted/test/test_sslverify.py`
    1. `test_allowedAnonymousClientConnection` shouldn't have a custom timeout
    1. Why is the `twisted.python.log` import moved inside `test_allowedAnonymousClientConnection`?
    1. I think the old version of `test_certificateOptionsSerialization` was actually better.  The useful assertion is "this object has no reference to that object".  Now the assertion is "this particular reference in this object is not a reference to that object", which is much weaker.  However, since it's a test for pickle support which I think we should give up pretending to offer, who cares?  Perhaps file a new ticket for deprecating these `__getstate__`/`__setstate__` and we can eventually delete them and this test.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst commented|
|-|-|

(In [36292]) Remove duplicate copyright, write better docstring for DistinguishedName. Refs [#6141](https://github.com/twisted/twisted/issues/6141)

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_6"></a>|@itamarst commented|
|-|-|

(In [36294]) Remove some debugging code that was left in by mistake. Refs [#6141](https://github.com/twisted/twisted/issues/6141)

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_7"></a>|@itamarst set owner to @exarkun|
|-|-|

  1. Done, in r36292 and r363293.
  2. Removed timeouts and logging code (just there for debugging) in r36294. Opened [#6166](https://github.com/twisted/twisted/issues/6166) for deprecating pickling support (I assume you're OK with leaving the pickling test in its current state).

Started new build: http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/sslverify-py3-6141-2

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_8"></a>|@itamarst commented|
|-|-|

Make that r36293.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set owner to @itamarst|
|-|-|

Great, thanks.  I am indeed okay with the current state of the pickling test.  Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_10"></a>|@itamarst set status to `closed`|
|-|-|

(In [36306]) Merge sslverify-py3-6141-2.

Author: itamar
Review: exarkun
Fixes: [#6141](https://github.com/twisted/twisted/issues/6141)

Port sslverify (in so far as test coverage allows) to Python 3.
